text/0000-rust-analyzer.md

```diff
@@ -0,0 +1,155 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The high-level plan is effectively to adopt rust-analyzer as the primary LSP implementation for the Rust project, and to aggressively pursue 'library-ification' as a means to eliminate code duplication. The ultimate vision is that the majority of the compiler logic should live in shared libaries which have two "front-ends", one from rustc and one from rust-analyzer.
+
+## Adopting rust-analyzer as the primary LSP implementation
+
+### Installing rust-analyzer today
+
+Today, [to install rust-analyzer for VSCode][r-a-install], one simply installs the rust-analyzer plugin. The plugin will download the rust-analyzer LSP implementation and automatically keep it up to date (currently on a once-per-week release cadence). The plugin also adds the `rust-src` component to rustup.
+
+[r-a-install]: https://rust-analyzer.github.io/manual.html#installation
+
+The experience of installing rust-analyzer for other editors is more varied. The rust-analyzer project only directly supports VSCode.  Other editor plugins are maintained independently.
+
+### Timeline for transition
+
+Transition will occur in three phases:
+
+* **Feedback:** During this first phase, we will post a blog post encouraging RLS users to try out rust-analyzer and see whether it works for them. If we encounter unexpected, blocking issues, or cases where people feel rust-analyzer is a significant step backward in their user experience, we may try to fix those issues before fully replacing the existing RLS.
+* **Deprecation period:** We announce that support for the RLS is deprecated. We begin putting in place the tooling to transition existing users away from the RLS.
+* **Final transition:** We no longer support the RLS plugin in its older form and no longer distribute RLS over rustup.
+
+### What is the transition plan?
+
+The precise transition plan is not part of this RFC. It will be determined and announced as we enter the Deprecation period, based on the feedback we've gotten and how many users have manually transitioned away from the RLS. We will endeavor to keep the experience as smooth as possible, but it may require some manual steps.
+
+### Branding: how to talk about rust-analyzer/RLS going forward?
+
+* We propose to keep the "rust-analyzer" name, at least for the transition period.
+* In keeping with the [proposed rust-lang github access policy](https://github.com/rust-lang/rfcs/pull/2872), the repositories from the [rust-analyzer github org](https://github.com/rust-analyzer) will be consolidated and then merged into the [rust-lang github org](https://github.com/rust-lang).
+    * They will be maintained by the compiler team with a dedicated working group.
+    * The infra team will work with rust-analyzer to integrate the binary release and upgrade process
```

Comment:
If the intent is to integrate `rust-analyzer` into the main release channels and handle it through `rustup` then at least some work will be needed in `rustup` (e.g. provision of proxy)